Algorithm/QCAlgorithm.cs

```diff
@@ -1858,5 +1890,14 @@ public string Download(string address, IEnumerable<KeyValuePair<string, string>>
                 return client.DownloadString(address);
             }
         }
+
+        /// <summary>
+        /// Event invocator for the <see cref="SignalsGenerated"/> event
+        /// </summary>
+        /// <param name="signals">The collection of signals generaed at the current time step</param>
+        protected virtual void OnSignalsGenerated(IEnumerable<Signal> signals)
+        {
+            SignalsGenerated?.Invoke(this, new SignalCollection(UtcTime, signals));
+        }
```

Comment:
Being declared `virtual`, if user algorithms `override` this method, they have to call `base.OnSignalGenerated()`.
Maybe better add a separate virtual method for users to override ?

Reply:
Hmm, in fact, maybe we just *don't* have this as virtual. Users can tap into the event stream via `IAlgorithm.SignalsGenerated += (sender, collection) => MyHandler(collection);` -- I don't see a need for them to insert themselves *before* our implementation runs.

Good call.